bndtools.builder/src/org/bndtools/builder/ComponentMarker.java

```diff
@@ -0,0 +1,285 @@
+
+        } catch (CoreException e) {
+            logger.logError("Component Marker error", e);
+        }
+    }
+
+    private static void parseChildrenForComponents(IPackageFragment pkg) throws JavaModelException, CoreException, BadLocationException {
+        for (IJavaElement e : pkg.getChildren()) {
+            if (e instanceof ICompilationUnit) {
+                ICompilationUnit compUnit = (ICompilationUnit) e;
+                if (!isComponentInImports(compUnit)) {
+                    continue;
+                }
+
+                compUnit.getResource().deleteMarkers(BndtoolsConstants.MARKER_COMPONENT, true, 1);
+                boolean hasComponent = hasComponentAnnotation(compUnit);
+                if (hasComponent) {
+
+                    beginASTParse(compUnit);
+                }
+            }
+        }
+    }
+
+    private static void beginASTParse(ICompilationUnit compUnit) throws CoreException, JavaModelException, BadLocationException {
+        ASTParser parser = ASTParser.newParser(AST.JLS8);
+        parser.setKind(ASTParser.K_COMPILATION_UNIT);
+        parser.setSource(compUnit);
+        parser.setResolveBindings(true);
+        CompilationUnit ast = (CompilationUnit) parser.createAST(null);
+        if (ast != null) {
+            ComponentAnnotationLocator visitor = new ComponentAnnotationLocator();
+            ast.accept(visitor);
+
+            Collection<Map.Entry<String,ISourceRange>> ranges = visitor.getSourceRange();
+            createComponentMarkers(compUnit, ranges);
+
+        }
+    }
+
+    private static boolean hasComponentAnnotation(ICompilationUnit c) throws CoreException, JavaModelException {
+        boolean hasComponent = false;
+        for (IType t : c.getTypes()) {
+            for (IAnnotation annot : t.getAnnotations()) {
+                if ("Component".equals(annot.getElementName())) {
+                    hasComponent = true;
+                    break;
+                }
+
+            }
+            if (hasComponent)
+                break;
+        }
+        return hasComponent;
+    }
+
+    private static void createComponentMarkers(ICompilationUnit c, Collection<Map.Entry<String,ISourceRange>> ranges) throws CoreException, JavaModelException, BadLocationException {
+        boolean componentFound = false;
+        for (Map.Entry<String,ISourceRange> entry : ranges) {
+            String message = entry.getKey();
+            ISourceRange range = entry.getValue();
+            IMarker marker = c.getResource().createMarker(BndtoolsConstants.MARKER_COMPONENT);
+            marker.setAttribute(IMarker.SEVERITY, IMarker.SEVERITY_INFO);
+            marker.setAttribute(IMarker.MESSAGE, message);
+            if (range != null) {
+
+                Document doc = new Document(c.getBuffer().getContents());
+
+                int line = doc.getLineOfOffset(range.getOffset()) + 1;
+                marker.setAttribute(IMarker.CHAR_START, range.getOffset());
```

Comment:
Do we need to parse the files at this level of detail?  can we make it faster by doing less? since we don't need to mark the char offsets in the .java file.